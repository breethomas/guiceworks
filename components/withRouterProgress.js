// @flow
import React, { PureComponent } from 'react'
import Router from 'next/router'

type State = {
  isRequesting: boolean,
}

export default (ComposedComponent: any) => (
  class extends PureComponent {
    state: State = { isRequesting: false }

    componentWillMount() {
      Router.onRouteChangeStart = () => this.setState({ isRequesting: true })
      Router.onRouteChangeComplete = () => this.setState({ isRequesting: false })
      Router.onRouteChangeError = () => this.setState({ isRequesting: false })
    }

    componentWillUnmount() {
      Router.onRouteChangeStart = null
      Router.onRouteChangeComplete = null
      Router.onRouteChangeError = null
    }

    render() {
      return (
        <ComposedComponent
          {...this.props}
          {...this.state}
        />
      )
    }
  }
)
