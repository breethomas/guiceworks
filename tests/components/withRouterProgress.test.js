import React from 'react'
import renderer from 'react-test-renderer'
import Router from 'next/router'
import withRouterProgress from '../../components/withRouterProgress'
import View from '../../components/View'

const ComposedComponent = withRouterProgress(View)

describe('withRouterProgress', () => {
  let tree

  beforeEach(() => {
    tree = renderer.create(<ComposedComponent data-passed="pass" />)
  })

  it('shows the correct snapshot for the wrapped ComposedComponent tree through state changes (normal)', () => {
    expect(tree.toJSON()).toMatchSnapshot()
    Router.onRouteChangeStart()
    expect(tree.toJSON()).toMatchSnapshot()
    Router.onRouteChangeComplete()
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('shows the correct snapshot for the wrapped ComposedComponent tree through state changes (error)', () => {
    expect(tree.toJSON()).toMatchSnapshot()
    Router.onRouteChangeStart()
    expect(tree.toJSON()).toMatchSnapshot()
    Router.onRouteChangeError()
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
