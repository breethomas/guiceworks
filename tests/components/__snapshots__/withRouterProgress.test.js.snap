// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRouterProgress shows the correct snapshot for the wrapped ComposedComponent tree through state changes (error) 1`] = `
<div
  data-passed="pass"
  isRequesting={false}
/>
`;

exports[`withRouterProgress shows the correct snapshot for the wrapped ComposedComponent tree through state changes (error) 2`] = `
<div
  data-passed="pass"
  isRequesting={true}
/>
`;

exports[`withRouterProgress shows the correct snapshot for the wrapped ComposedComponent tree through state changes (error) 3`] = `
<div
  data-passed="pass"
  isRequesting={false}
/>
`;

exports[`withRouterProgress shows the correct snapshot for the wrapped ComposedComponent tree through state changes (normal) 1`] = `
<div
  data-passed="pass"
  isRequesting={false}
/>
`;

exports[`withRouterProgress shows the correct snapshot for the wrapped ComposedComponent tree through state changes (normal) 2`] = `
<div
  data-passed="pass"
  isRequesting={true}
/>
`;

exports[`withRouterProgress shows the correct snapshot for the wrapped ComposedComponent tree through state changes (normal) 3`] = `
<div
  data-passed="pass"
  isRequesting={false}
/>
`;
